test/test_katello/test_deb_tracer.py

```diff
@@ -0,0 +1,176 @@
+import unittest2 as unittest
+from katello.deb_tracer import (
+    needrestart,
+    has_needrestart,
+    use_flag,
+    use_needrestart,
+    use_needrestart,
+    AptTracerApp,
+    collect_apps
+)
+import subprocess
+import sys
+
+from mock import patch, Mock
+
+
+class TestNeedrestart(unittest.TestCase):
+
+    @patch.object(subprocess, "check_output")
+    def test_needrestart_output(self, m):
+        needrestart_output = """
+NEEDRESTART-VER: 3.5
+NEEDRESTART-KCUR: 5.10.0-7-amd64
+NEEDRESTART-KEXP: 5.10.0-10-amd64
+NEEDRESTART-KSTA: 3
+NEEDRESTART-SVC: cron.service
+NEEDRESTART-SVC: dbus.service
+NEEDRESTART-SVC: getty@tty1.service
+NEEDRESTART-SVC: ifup@ens160.service
+NEEDRESTART-SVC: rsyslog.service
+NEEDRESTART-SVC: ssh.service
+NEEDRESTART-SVC: systemd-journald.service
+NEEDRESTART-SVC: systemd-logind.service
+NEEDRESTART-SVC: systemd-timesyncd.service
+NEEDRESTART-SVC: unattended-upgrades.service
+NEEDRESTART-SVC: user@0.service """
+
+        output_list = needrestart_output.split("\n")
+        m.return_value = needrestart_output
+        self.assertEqual(needrestart(), output_list)
+        self.assertEqual(m.call_args.args, (["needrestart", "-b"],))
+        self.assertEqual(m.call_args.kwargs, {"universal_newlines": True})
+        self.assertEqual(len(collect_apps()), 12)
+
+    @patch.object(subprocess, "check_output")
+    def test_needrestart_exception_file_not_found(self, m):
+        m.side_effect = OSError()
+        with self.assertRaises(SystemExit) as cm:
+            needrestart()
+        the_exception = cm.exception
+        self.assertEqual(the_exception.code, 1)
+
+
+class TestCollectApps(unittest.TestCase):
+    @unittest.skipIf(sys.version_info[0] == 3, "not compatible with Python 3")
```

Comment:
Is this still true now that you have the Python 2 compatibility code in there? If it is still incompatible, can't you apply the `skipIf` on the entire class instead?

Reply:
I removed the two of the four tests, because we do not need the distinction between python2 and python3 anymore, because of your smart suggestion to name the import of the distutils.spawn as the import of shutil.